MessageTerminal/MessageTerminal.cxx

```diff
@@ -0,0 +1,143 @@
+#include <unistd.h>
+
+#include <iostream>
+#include <sstream>
+
+#include "LinuxSerialDevice.h"
+#include "TestStandCommHost.h"
+
+#define BAUD_RATE 115200
+
+#define BASIC_CMD(_name)                    \
```

Comment:
Just looking to understand this structure. It is a macro where `_name` is replaced by the function `void _name(...)`?

Reply:
It gets replaced with the entire function, so:

`BASIC_CMD(ping);`

gets replaced by exactly this:

```c
void ping(istringstream& iss)
{
    if (comm.ping()) cout << "OK" << endl;
    else cout << "ERROR" << endl;
};
```